kronolith/lib/Application.php

```diff
@@ -945,6 +945,35 @@ public function davGetCollections($user)
                 '{http://sabredav.org/ns}read-only' => !$share->hasPermission($hordeUser, Horde_Perms::EDIT),
             );
         }
+        // External Calendars from other Horde Apps
+                foreach ($calendar_manager->get(Kronolith::ALL_EXTERNAL_CALENDARS) as $calendarId => $calendar) {
+            // We don't want to duplicate tasks handling
+            if ($calendar->api() == 'tasks') {
+                continue;
+            }
+            try {
+                $id = $dav->getExternalCollectionId($calendar->internalId(), 'calendar');
+            } catch (Horde_Dav_Exception $e) {
+            }
+            $calendars[] = array(
+                'id' => $id,
+                'uri' => $id,
+                '{' . CalDAV\Plugin::NS_CALENDARSERVER . '}shared-url' =>
+                    $calendar->caldavUrl(),
+                'principaluri' => 'principals/' . $user,
+                '{http://sabredav.org/ns}owner-principal' =>
+                    'principals/-system-', // TODO: Not sure who should really own these, system or the calling user
```

Comment:
I tend to own this to the user. In fact we have both, global and per-user resourced provided through the timeobjects API. But it makes more sense IMO to have different per-user calendars with the same data, than single global calendars with individual data.